Intro/README.md

```diff
@@ -0,0 +1,455 @@
+virtual, and simulated TPMs as well, all implemented in software.
+
+To simplify things we'll consider only TPM 2.0.  Also to simplify things
+we'll ignore algorithm agility.
+
+Other parts of this [tutorial](README.md) may cover specific concepts in
+much more detail.
+
+# Core Concepts
+
+Some core concepts in the world of TPMs
+
+ - cryptography:
+    - digest functions
+    - hash extension
+    - HMAC
+    - symmetric encryption
+    - asymmetric encryption and/or signature algorithms
+ - platform configuration registers (PCRs)
+ - cryptographic object naming
+ - immutability of object public areas
+ - key hierarchies
+ - key wrapping
+ - limited resources
+    - tickets
+    - resource management
+ - sessions
+ - authorization
+    - restricted cryptographic keys
+    - policies
+ - other object types
+    - non-volatile (NV) indexes
+ - attestation
+
+We'll assume reader familiarity with cryptography so we need not explain
+it.
+
+Authorization is the most important aspect of a TPM, since that's
+ultimately what it exists for: to authorize a system or application to
+perform certain duties when all the desired conditions allow for it.
+
+TPMs have a very rich set of options for authorization.  It's not just
+[policies](#Policies), but also cryptographic object names used with
+restricted keys to allow access only to applications that also have
+other access.
+
+Where to start?  Let's start with hash extension, which may be the only
+trivial concept in the world of TPMs!
+
+## Important Notes
+
+ - Chained implications.
+
+   It is important to note the TPM ecosystem's critical dependence on
+   TPMs performing authorization as specified.  A lot can be
+   accomplished by chaining cryptographic implications.  Sending a
+   secret to a TPM-using application, with the secret encrypted in that
+   TPM's primary endorsement public key is safe when one knows that
+   public key truly is a legitimate hardware TPM's public key: the TPM
+   will not release the secret to the application unless the recipient
+   has some access specified by the sender.  Complex chains of events
+   can be secure provided one knows that it involves legitimate hardware
+   TPMs that otherwise could not be secured.  Therefore the legitimacy
+   of a TPM that one communicates with is essential.
+
+ - Limited resources.
+
+   Hardware TPMs are very resource-starved.  The set of commands defined
+   by the TCG TPM specifications is specifically constructed to a) be
+   small and simple (simple-enough anyways), yet b) enable rich
+   semantics in spite of this.
+
+   These limited resources mean, among other things, that TPMs don't do
+   complex things like PKIX certificate validation path construction, or
+   even PKIX certificate path validation, nor do they implement complex
+   authentication protocols using PKIX, nor Kerberos, nor even simple
+   protocols like JWT.  Hardware TPMs can be used to implement portions
```

Comment:
I'm not sure in an intro doc this long list of things the TPM can't do is necessary, particularly as some of them are pretty esoteric.

Reply:
Perhaps, yeah.  I guess what I want to say is: TPMs only do very basic things because they are very limited.

Reply:
Maybe also mentioning that the TPM is a passive device? Including that a TPM only responds to request from the Host rather than act on its own.

Reply:
Removed.